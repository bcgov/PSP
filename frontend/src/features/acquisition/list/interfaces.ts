export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
