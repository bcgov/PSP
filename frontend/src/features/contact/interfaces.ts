export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
