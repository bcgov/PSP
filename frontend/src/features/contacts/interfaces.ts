export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}

export enum ContactTypes {
  ORGANIZATION = 'O',
  INDIVIDUAL = 'P',
}
