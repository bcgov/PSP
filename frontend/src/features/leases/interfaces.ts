export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
