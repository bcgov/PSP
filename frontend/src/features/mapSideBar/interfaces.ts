import { AddressTypes, PropertyTypes } from 'constants/index';
import { IAddress } from 'interfaces';

export const defaultPropertyValues = {
  address: {
    addressTypeId: AddressTypes.Physical,
    streetAddress1: '',
    municipality: '',
    postal: '',
    province: '',
    provinceId: 1,
    region: '',
    district: '',
  } as IAddress,
  pid: '',
  propertyTypeId: PropertyTypes.Land,
  classification: '',
  region: '',
  district: '',
  zoning: '',
  landArea: '',
  landLegalDescription: '',

  // TODO: Theses properties probably should be replaced or updated.
  titleNumber: '',
  purpose: '',
  ruralArea: '',
  ocpDesignation: '',
};
