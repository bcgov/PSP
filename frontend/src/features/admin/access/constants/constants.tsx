import { ColumnWithProps } from 'components/Table';
import { AccessRequestStatus, AccessStatusDisplayMapper } from 'constants/accessStatus';
import { AccessRequestForm } from 'features/admin/access-request/models';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

import { RowActions } from '../components/RowActions';

export const getAccessRequestColumns = (
  refresh: () => void,
): ColumnWithProps<AccessRequestForm>[] => [
  {
    Header: 'IDIR/BCeID',
    accessor: 'businessIdentifierValue',
    align: 'left',
    Cell: (props: CellProps<AccessRequestForm>) => {
      return (
        <Link to={`/admin/access/request/${props.row.original.id}`}>
          {props.row.original.businessIdentifierValue}
        </Link>
      );
    },
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
  },
  {
    Header: 'Last name',
    accessor: 'surname',
    align: 'left',
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    minWidth: 200,
    maxWidth: 300,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
  },
  {
    Header: 'Status',
    accessor: 'accessRequestStatusTypeCodeId',
    align: 'left',
    width: 100,
    Cell: (props: CellProps<AccessRequestForm>) => {
      return AccessStatusDisplayMapper[
        props.row.original.accessRequestStatusTypeCodeId as AccessRequestStatus
      ];
    },
  },
  {
    Header: 'Role',
    accessor: 'roleName',
    align: 'left',
    minWidth: 200,
  },
  {
    Header: 'MoTI Region',
    accessor: 'regionName',
    align: 'left',
    minWidth: 200,
  },
  {
    Header: 'Actions',
    Cell: (props: CellProps<AccessRequestForm>) => {
      return <RowActions {...props} refresh={refresh} />;
    },
    width: 75,
  },
];
