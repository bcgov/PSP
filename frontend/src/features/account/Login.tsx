import * as actionTypes from 'constants/actionTypes';
import { useQuery } from 'hooks/use-query';
import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Row from 'react-bootstrap/Row';
import Spinner from 'react-bootstrap/Spinner';
import { FaExternalLinkAlt } from 'react-icons/fa';
import { Redirect } from 'react-router-dom';
import { useAppSelector } from 'store/hooks';
import { NEW_PIMS_USER } from 'store/slices/users';
import { Logo, useTenant } from 'tenants';

import { LoginStyled } from './LoginStyled';

/**
 * Login component provides information and links for the user to login.
 * @returns Login component.
 */
const Login = () => {
  const { redirect } = useQuery();
  const [showInstruction, setShowInstruction] = useState(false);
  const keyCloakWrapper = useKeycloakWrapper();
  const keycloak = keyCloakWrapper.obj;
  const isIE = usingIE();
  const activated = useAppSelector(state => state.network[actionTypes.ADD_ACTIVATE_USER]);
  const tenant = useTenant();

  if (!keycloak) {
    return <Spinner animation="border"></Spinner>;
  }
  if (keycloak?.authenticated) {
    if (activated?.status === NEW_PIMS_USER || !keyCloakWrapper?.roles?.length) {
      return <Redirect to={{ pathname: '/access/request' }} />;
    }
    return <Redirect to={redirect || '/mapview'} />;
  }
  if (isIE) {
    return <Redirect to={{ pathname: '/ienotsupported' }} />;
  }

  return (
    <LoginStyled className="login" fluid={true}>
      <Container className="unauth" fluid={true}>
        <Logo withText={true} />
        <Row className="sign-in">
          <Col xs={1} md={3} />
          <Col xs={16} md={6} className="block">
            <h1>{tenant.login.title}</h1>
            <h6>{tenant.login.heading}</h6>
            <p>{tenant.login.body}</p>
            <Button variant="primary" onClick={() => keycloak.login()}>
              Sign In
            </Button>
            <p>Sign into PIMS with your government issued IDIR or with your Business BCeID.</p>
            <Row className="bceid">
              <Button variant="link" onClick={() => setShowInstruction(!showInstruction)}>
                Don't have a Business BCeID?
              </Button>
            </Row>
            <Row>
              {showInstruction && (
                <Jumbotron>
                  <p>
                    1. Search to see if your entity is{' '}
                    <a
                      href="https://www.bceid.ca/directories/whitepages"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      already registered
                    </a>{' '}
                    <FaExternalLinkAlt />
                  </p>
                  <p>
                    If you're not yet registered, <br></br>2.{' '}
                    <a
                      href="https://www.bceid.ca/os/?7169"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      Register for your Business BCeID
                    </a>{' '}
                    <FaExternalLinkAlt />
                  </p>
                </Jumbotron>
              )}
            </Row>
          </Col>
          <Col xs={1} md={3} />
        </Row>
      </Container>
    </LoginStyled>
  );
};

//check to see if user is using Internet Explorer
//as their browser
const usingIE = () => {
  const userAgent = window.navigator.userAgent;
  const isOldIE = userAgent.indexOf('MSIE '); //tag used for IE 10 or older
  const isIE11 = userAgent.indexOf('Trident/'); //tag used for IE11
  if (isOldIE > 0 || isIE11 > 0) return true;
  return false;
};

export default Login;
