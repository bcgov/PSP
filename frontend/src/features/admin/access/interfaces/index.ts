import { AccessRequestStatus } from 'constants/accessStatus';

export interface IAccessRequestModel {
  id: number;
  userId: number;
  businessIdentifierValue: string;
  firstName: string;
  surname: string;
  email: string;
  position: string;
  status: AccessRequestStatus;
  organization: string;
  role: string;
  note: string;
}
