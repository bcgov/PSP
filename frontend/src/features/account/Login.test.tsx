import { useKeycloak } from '@react-keycloak/web';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { ADD_ACTIVATE_USER } from 'constants/actionTypes';
import { createMemoryHistory, MemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureMockStore, { MockStoreEnhanced } from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IGenericNetworkAction } from 'store/slices/network/interfaces';
import { networkSlice } from 'store/slices/network/networkSlice';
import { ThemeProvider } from 'styled-components';
import { TenantConsumer, TenantProvider } from 'tenants';

import Login from './Login';

jest.mock('axios');
jest.mock('@react-keycloak/web');
const mockStore = configureMockStore([thunk]);

const defaultStore = mockStore({
  [networkSlice.name]: {
    [ADD_ACTIVATE_USER]: {},
  },
});

const TestLogin = ({
  history,
  store,
}: {
  history: MemoryHistory<unknown>;
  store?: MockStoreEnhanced<unknown, {}>;
}) => {
  return (
    <TenantProvider>
      <TenantConsumer>
        {({ tenant }) => (
          <ThemeProvider theme={{ tenant, css: {} }}>
            <Provider store={store ?? defaultStore}>
              <Router history={history}>
                <Login />
              </Router>
            </Provider>
          </ThemeProvider>
        )}
      </TenantConsumer>
    </TenantProvider>
  );
};

//boilerplate function used by most tests to wrap the Login component with a router.
const renderLogin = () => {
  process.env.REACT_APP_TENANT = 'CITZ';
  const history = createMemoryHistory();
  return render(<TestLogin history={history} />);
};

describe('login', () => {
  afterEach(() => {
    cleanup();
  });
  it('login renders correctly', () => {
    (useKeycloak as jest.Mock).mockReturnValue({ keycloak: { authenticated: false } });
    process.env.REACT_APP_TENANT = 'MOTI';
    const history = createMemoryHistory();
    const tree = renderer.create(<TestLogin history={history} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('authenticated users are redirected to the mapview', () => {
    process.env.REACT_APP_TENANT = 'MOTI';
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: { authenticated: true, userInfo: { groups: ['System Administrator'] } },
    });
    const history = createMemoryHistory();
    render(<TestLogin history={history} />);
    expect(history.location.pathname).toBe('/mapview');
  });

  it('new users are sent to the guest page', () => {
    process.env.REACT_APP_TENANT = 'CITZ';
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: { authenticated: true, realmAccess: { roles: [{}] } },
    });
    const history = createMemoryHistory();
    const activatedAction: IGenericNetworkAction = {
      isFetching: false,
      name: 'test',
      type: 'POST',
      status: 201,
    };
    const store = mockStore({
      [networkSlice.name]: {
        activateUser: activatedAction,
      },
    });

    render(<TestLogin history={history} store={store} />);
    expect(history.location.pathname).toBe('/access/request');
  });

  it('unAuthenticated users are shown the login screen', () => {
    (useKeycloak as jest.Mock).mockReturnValue({ keycloak: { authenticated: false } });
    const { getAllByRole } = renderLogin();
    expect(getAllByRole('heading')[0]).toHaveTextContent(
      'Search and visualize government property information',
    );
    expect(getAllByRole('heading')[1]).toHaveTextContent(
      'PIMS enables you to search properties owned by the Government of British Columbia',
    );
  });

  it('a spinner is displayed if keycloak has not yet been initialized', () => {
    (useKeycloak as jest.Mock).mockReturnValue({ keycloak: undefined });
    const { container } = renderLogin();
    expect(container.firstChild).toHaveClass('spinner-border');
  });

  it('the login button calls keycloaks login() method', () => {
    const login = jest.fn();
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: { login: login, authenticated: false },
    });

    const { getByText } = renderLogin();
    fireEvent.click(getByText(/Sign In/));

    expect(login.mock.calls.length).toBe(1);
  });
});
