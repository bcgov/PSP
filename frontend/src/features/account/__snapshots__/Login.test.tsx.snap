// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login login renders correctly 1`] = `
.c0 {
  position: relative;
  background-image: url("/tenants/MOTI/background-image.jpg");
  background-size: cover;
  background-position: center;
  overflow: auto;
  grid-area: content;
}

.c0 .btn.btn-primary {
  display: inline-block;
}

.c0 .btn.btn-primary {
  display: inline-block;
}

.c0 .unauth {
  font-size: 2.4rem;
  margin: 9.1rem auto;
  background-color: rgba(250,250,250,0.9);
  padding-top: 1.5rem;
  padding-bottom: 2rem;
  max-width: 75rem;
  border-radius: 0.4rem;
}

.c0 .unauth .sign-in {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .unauth .block {
  padding: 2rem 0;
  max-width: 65rem;
  max-height: 47.5rem;
  text-align: center;
  border-radius: 0.4rem;
  min-width: 90%;
}

.c0 .unauth h1 {
  padding-bottom: 1.8rem;
  font-family: 'BCSans',Fallback,sans-serif;
  font-weight: 700;
  font-size: 2.4rem;
  font-style: italic;
}

.c0 .unauth h6 {
  margin-bottom: 2rem;
}

.c0 .unauth p {
  font-size: 1.2rem;
  font-family: 'BCSans',Fallback,sans-serif;
  line-height: 130%;
  margin-left: 3rem;
  margin-right: 3rem;
  margin-bottom: 0.2rem;
}

.c0 .unauth .btn {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.c0 .unauth .btn-link {
  margin-left: 3rem;
  margin-top: 1rem;
  font-size: 1.2rem;
}

.c0 .unauth .btn-link:focus,
.c0 .unauth .btn-link:active {
  background: none !important;
}

.c0 .unauth .bceid {
  margin-bottom: -3rem;
}

.c0 .unauth .jumbotron {
  background-color: white;
  padding: 2rem;
  margin-left: 4rem;
  margin-right: 4rem;
  width: 90%;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  border: 0.1rem;
  border-style: solid;
  border-color: #707070;
  text-align: left;
}

.c0 .unauth .jumbotron p {
  margin: 1rem;
}

.c0 .unauth .blockText {
  margin: 0 auto 3.8rem auto;
}

.c0 .unauth .or {
  margin: 2rem auto;
}

.c0 .unauth .border-dark {
  border-color: !important;
  color: !important;
}

.c0 .unauth .pims-logo {
  margin-bottom: 1rem;
}

<div
  className="c0 login container-fluid"
>
  <div
    className="unauth container-fluid"
  >
    <img
      alt="PIMS logo"
      className="pims-logo"
      src="/tenants/MOTI/PIMS-logo-with-text.png"
    />
    <div
      className="sign-in row"
    >
      <div
        className="col-md-3 col-1"
      />
      <div
        className="block col-md-6 col-16"
      >
        <h1>
          TRAN Property Information Management System (PIMS)
        </h1>
        <h6>
          PIMS enables you to view highways and properties owned by the Ministry of Transportation and Infrastructure
        </h6>
        <p>
          WARNING: Not all data included within has been vetted for accuracy and completeness. Please use caution when proceeding and confirm data before relying on it.
        </p>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Sign In
        </button>
        <p>
          Sign into PIMS with your government issued IDIR or with your Business BCeID.
        </p>
        <div
          className="bceid row"
        >
          <button
            className="btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Don't have a Business BCeID?
          </button>
        </div>
        <div
          className="row"
        />
      </div>
      <div
        className="col-md-3 col-1"
      />
    </div>
  </div>
</div>
`;
